src/lsp/sql_parser/grammar.mly

```diff
@@ -0,0 +1,134 @@
+(*For commit*)
+%token COMMIT
+(*For SQL*)
+%token ASTERIX SELECT FROM
+%token <string> WORD
+%token <string> STRING
+%token <string> DIGITS
+%token <string> COBOL_VAR
+%token <string> BACKSLASHED_VAR
+%start <esql_instuction> main 
+
+%%
+
+let main :=
+| EXEC; SQL; stm=esql ; END_EXEC; EOF; {stm}
+
+let variable := 
+| s=WORD; {SqlVar s} 
+| s=COBOL_VAR; {CobolVar s} 
+
+let esql := 
+| AT; v=variable; stm=esql; {At(v, stm)}
+| BEGIN; DECLARE; SECTION; {BeginDeclare}
+| END; DECLARE; SECTION; {EndDeclare}
+| START; TRANSACTION; {StartTransaction}
+| WHENEVER; c=whenever_condtion; k= whenever_continuation; {Whenever(c, k)}
+| i = sql; {Sql(i)}
+| INCLUDE; i=includeStm; {Include(i)}
+| CONNECT; c=connectStm; {Connect(c)}
+| ROLLBACK; r=rb_work_or_tran; a=rb_args; {Rollback(r, a)}
+| COMMIT; wt= option(rb_work_or_tran); RELEASE; {Commit(wt, true)}
+| COMMIT; wt= option(rb_work_or_tran); {Commit(wt, false)}
+| SAVEPOINT; s=variable; {Savepoint(s)}
+
+let rb_work_or_tran :=
+| WORK; {Work}
+| TRANSACTION; {Transaction}
+
+let rb_args :=
+| RELEASE; {Release}
+| TO; SAVEPOINT; v = variable; {To(v)}
+| TO; v = variable; {To(v)} 
+
+let connectStm :=
+/*
+EXEC SQL CONNECT TO :dbname [ AS :db_conn_id ] 
+USER :username USING :db_data_source 
+IDENTIFIED BY :password 
+*/
+| TO; dbname= variable; db_conn_id=option(as_var); USER; username=variable;
+  USING; db_data_source= variable; IDENTIFIED; BY; password= variable;
+  { Connect_to_idby {dbname; db_conn_id; username; db_data_source; password} }
+
+/* 
+EXEC SQL CONNECT TO :db_data_source [ AS :db_conn_id ]
+USER :username.:opt_password [ USING password ];
+-> Supporté si il n'y as pas de opt_passwod
+ */
+| TO; db_data_source= variable; db_conn_id=option(as_var); USER; username=variable;
+  password = option(using_var);
+  { Connect_to {db_data_source; db_conn_id; username; password} }
+/* 
+EXEC SQL CONNECT USING :db_data_source 
+(credentials must be embedded to be able to connect) 
+*/
+| USING; db_data_source=variable; {Connect_using{db_data_source}}
+/* 
+EXEC SQL CONNECT :username IDENTIFIED BY :password 
+[ AT :db_conn_id ] [ USING :db_data_source] (mode 4 is unsupported)
+ */
+| username= variable; IDENTIFIED; BY; password= variable; 
+  db_conn_id = option(at_var); db_data_source= option(using_var);
+  {Connect_user{username; password; db_conn_id; db_data_source}}
+| RESET; name=option(variable); 
+  {Connect_reset(name)}
+
+let at_var:= AT; p=variable; {p}
+
+let using_var:= USING; p=variable; {p}
+
+let as_var:= AS; v=variable; {v} 
+
+let includeStm :=
+| SQLCA; {Sqlca}
+| SQLDA; {Sqlda}
+| s = variable; {Filename(s)}
+
+let whenever_condtion :=
+| NOT; FOUND; {Not_found}
+| SQLERROR; {SqlError}
+| SQLWARNING; {SqlWarning}
+
+let whenever_continuation :=
+| CONTINUE;  {Continue}
+| PERFORM; label=variable; {Perform (label)}
+| GOTO; stmt_label=variable; {Goto (stmt_label)}
+
+(*SQL Stuff*)
+let sql :=
+| SELECT; prj = projection; FROM; src = source ; {Select (prj, src)}
```

Comment:
We have to discuss how to deal with SELECT.

Reply:
Yes, this was temporary, to have some SQL in the ast